server/plugin.ts

```diff
@@ -72,7 +87,7 @@ export class AnomalyDetectionOpenSearchDashboardsPlugin
     );
 
     // Create services & register with OpenSearch client
-    const adService = new AdService(client);
+    const adService = new AdService(client, dataSourceEnabled);
```

Comment:
How will the `adService` behave in MDS world since `client` will be `undefined` here? I checked the `ad.ts` file and `client` is used as a `Local Cluster`.

Reply:
in MDS world, the open search legacy client will be undefined and it will pickup the data source client as in MDS world dataSourceEnabled will always be true